superset-frontend/src/explore/reducers/datasourcesReducer.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Dataset } from '@superset-ui/chart-controls';
+import { getDatasourceUid } from 'src/utils/getDatasourceUid';
```

Comment:
wdyt about naming this dataset instead of datasource? I feel that datasource is very general, and have been trying to move away from using it when possible when referring to specifically datasets.

Reply:
On second thought, it looks like this could be used for other datasource types, in which case it works to keep this named 'datasource'